### ISSUE TITLE
[Question] how to update button status behind the barrier ?

### ISSUE THREAD

**Issue by fvisticot** [KlausJokisuo/modals#11]
I have a settings button used to launch a modal with this plugin.
I need to close the modal when click is done outside the modal (using the barrierDismissible=true and no color to see the settings button)

As the barrier is on top of the settings button how to get informed if the user clicks on the settings button again to close the dialog ?
When the click is done on the settings button, the modal is closed (because outside the dialog) but the button onPressed callback is not triggered

I have tested using the onRemove callback and trying to update button state with a dedicated controller but it seems I can not execute a "setState" inside this onRemove callback

Error I get:

```
══╡ EXCEPTION CAUGHT BY FOUNDATION LIBRARY ╞════════════════════════════════════════════════════════
The following assertion was thrown while dispatching notifications for AppBarMenuButtonController:
setState() or markNeedsBuild() called when widget tree was locked.
This AppBarMenuButton widget cannot be marked as needing to build because the framework is locked.
The widget on which setState() or markNeedsBuild() was called was:
  AppBarMenuButton
```

![image](https://user-images.githubusercontent.com/153679/147853127-46253478-5933-4d10-b5c3-15f4e35bf92c.png)

**Comment by KlausJokisuo** [comment]
Thanks for opening an issue.
I was able to reproduce the problem using the setState within the onRemove callback. This issue has been fixed on version 0.0.3.

Let me know if your issue is fixed by the new release :)

**Comment by fvisticot** [comment]
> @fvisticot Thanks for opening an issue.
> 
> I was able to reproduce the problem using the setState within the onRemove callback. This issue has been fixed on version 0.0.3.
> 
> Let me know if your issue is fixed by the new release 🙂

Tx a lot !!! works perfectly

**Comment by KlausJokisuo** [comment]
Great to hear! I'm closing the issue because it has been solved. Don't hesitate to create a new one if more issues appear.
